tests/test_urls.py

```diff
@@ -119,15 +121,17 @@ def test_missing_parameters_for_new_issue(self):
 
     def test_new_issue_should_not_crash(self):
         '''Checks 500 is not accepted for /issues/new POST.'''
-        data = {'problem_category': u'mobile_site_bug',
-                'username': u'',
-                'description': u'foo',
-                'submit-type': u'github-proxy-report',
-                'url': u'http://example.com',
-                'os': u'Foobar',
-                'browser': u'BarFoo'}
-        rv = self.app.post('/issues/new', data=data)
-        self.assertNotEqual(rv.status_code, 500)
+        with patch('requests.post') as github_data:
+            github_data.status_code.return_value = 500
+            data = {'problem_category': u'mobile_site_bug',
+                    'username': u'',
+                    'description': u'foo',
+                    'submit-type': u'github-proxy-report',
+                    'url': u'http://example.com',
+                    'os': u'Foobar',
+                    'browser': u'BarFoo'}
+            rv = self.app.post('/issues/new', data=data)
+            self.assertNotEqual(rv.status_code, 500)
```

Comment:
ah interesting. Replacing the test and running them

```bash
nosetests --verbose --nocapture tests/test_urls.py
```

We get:


```
Test that /about exists. ... ok
Test that asks user to log in before displaying activity. ... ok
Test POST to /csp-report w/ correct content-type returns 204. ... ok
Test POST w/ wrong content-type to /csp-report returns 400. ... ok
Test that the home page exists. ... ok
Test issues and integer for: ... ok
Test that the /issues/<number> exists, and does not redirect. ... ok
Test that the /issues route gets 200 and does not redirect. ... ok
Sends 400 to POST on /issues/new with missing parameters. ... ok
Test that /issues/new exists. ... ok
Checks 500 is not accepted for /issues/new POST. ... --------------------------------------------------------------------------------
INFO in views [/Users/karl/code/webcompat.com/webcompat/views.py:206]:
127.0.0.1 http://example.com
--------------------------------------------------------------------------------
FAIL
Test that /privacy exists. ... ok
Rate Limit URI sends 410 Gone. ... ok
Test that the /tools/cssfixme route gets 200. ... ok
Test that the /tools/cssfixme route gets 200 with ?url query. ... ok
Test that the /tools/cssfixme route gets 200 with bad ?url query. ... ok

======================================================================
FAIL: Checks 500 is not accepted for /issues/new POST.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/karl/code/webcompat.com/tests/test_urls.py", line 130, in test_new_issue_should_not_crash
    self.assertEqual(rv.status_code, 400)
AssertionError: 302 != 400
-------------------- >> begin captured logging << --------------------
requests.packages.urllib3.connectionpool: DEBUG: Starting new HTTPS connection (1): api.github.com
requests.packages.urllib3.connectionpool: DEBUG: https://api.github.com:443 "POST /repos/webcompat/webcompat-tests/issues HTTP/1.1" 201 2017
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 16 tests in 1.642s

FAILED (failures=1)
```

we get a 302! Let's explore a bit more.